webapp/tests/karma/ts/effects/reports.effects.spec.ts

```diff
@@ -140,6 +140,41 @@ describe('Reports effects', () => {
       expect(unsetSelected.callCount).to.equal(0);
     });
 
+    it('should load reports when selected nearly simultaneously', async () => {
+      const setLoadingShowContent = sinon.stub(GlobalActions.prototype, 'setLoadingShowContent');
+      const setSelected = sinon.stub(ReportsActions.prototype, 'setSelected');
+      const unsetSelected = sinon.stub(GlobalActions.prototype, 'unsetSelected');
+      reportViewModelGeneratorService.get.onFirstCall().resolves({_id: 'reportID0', model: true});
+      reportViewModelGeneratorService.get.onSecondCall().resolves({_id: 'reportID1', model: true});
+
+      actions$ = of(ReportActionList.selectReport({id: 'reportID0', silent: true}),
```

Comment:
Does this mean that the report is selected immediately after the other? 

This is about interaction with report verification. 
So your unit test should be like: 
a. One report is completely selected (otherwise you can't verify in the UI)
b. Report is verified
c. While report verification is "pending" (promise chain not finished) select another report.

Reply:
Yeah, sorry. I was going to have you take a look at this, but then I decided that I actually had a few other things I wanted to tinker with first. But, apparently GH still notified you even though I removed you as a reviewer. :confused: Oh well, thanks for taking a look!

Anyway, my understanding of this line is that, yes, both selections are emitted one-immediately-after-the-other. That was my desired behavior for this version of the test as it helped me to actually understand the behavioral differences of the different *Map functions, but as you mentioned, there is some other higher-order behavior that the test should also probably include....

So, yes, I am still working on putting together a more comprehensive workflow test like the one you mention.